src/reducers/language.js

```diff
@@ -0,0 +1,88 @@
+import { CHANGE_CONST, CHANGE_FUN, CHANGE_PRED } from '../actions';
+import {parseConstants, parseFunctions, parsePredicates} from '@fmfi-uk-1-ain-412/js-fol-parser';
+
+const language = (state = { const: { input: "", object: new Set(), error: ""}, fun: { input: "", object: new Map(), error: "" }, pred: { input: "", object: new Map(), error: "" } }, action = {type: undefined}) => {
+    switch (action.type) {
+        case CHANGE_CONST:
+            let newState = {
+                const: {
+                    input: action.text,
+                    object: state.const.object,
+                    error: state.const.error
+                },
+                fun: state.fun,
+                pred: state.pred
+            }
+            try {
+                let res = undefined;
+                if (action.text !== "") {
+                    res = parseConstants(action.text); 
+                }
+                newState.const.object = new Set(res);
+                newState.const.error = "";
+            } catch (e) {
+                newState.const.error = "<b>" + action.text.substring(0, e.location.start.offset) + "<mark class='text-danger'>" +
+                action.text.substring(e.location.start.offset, e.location.end.offset) + "</mark>" +
+                action.text.substring(e.location.end.offset, action.text.length) + "</b><br/>" +
+                e.name + ": " + e.message;
+            }

```

Comment:
Generovanie HTML sa ma diat v komponentoch. Tu si len ulozte do stavu data, ktore na to budete potrebovat.